course02/homework01/danielellis/linkedlist.cc

```diff
@@ -0,0 +1,86 @@
+#include "linkedlist.h"
+
+#include <stdexcept>
+
+using namespace std;
+
+LinkedList::LinkedList() : mSize(0)
+{
+}
+
+LinkedList::LinkedList(const LinkedList& ll)
+{
+	if (ll.Size() > 0)
+	{
+		shared_ptr<LinkedListNode> last;
+		mStart = ll.mStart->Clone(last);
+		mEnd = last;
+	}
+	mSize = ll.Size();
```

Comment:
This copy ctor is too complicated. It seems that the responsibility of cloning should be here, and not in the Node. Having it in the Node complicated things. It should be a simple for loop.

Reply:
Note that the for loop would at the same time remove this if !